src/app/dataset_storage.py

```diff
@@ -1,12 +1,53 @@
 import json
 
 from google.cloud import storage
+from logging_config import logging
 
+logger = logging.getLogger(__name__)
 storage_client = storage.Client()
 
 
 def get_dataset(filename, bucket_name):
-    """Used by the cloud function."""
+    """
+    Used by the cloud function.
+    * Process if the file content is proper JSON syntax or else log error.
+    """
     bucket = storage_client.bucket(bucket_name)
-    dataset = json.loads(bucket.blob(filename).download_as_string())
-    return dataset
+    try:
+        dataset = json.loads(bucket.blob(filename).download_as_string())
+        isValid, missing_keys = validate_keys_dataset(dataset)
+        if isValid is True:
+            return dataset
+        else:
+            message = ""
+            for key in missing_keys:
+                message = message + key + ", "
+            logger.error(f"The keys {message} are missing in the JSON object")
+            return None
+    except ValueError as e:
+        logger.error(f"Invalid JSON in the file {filename} - %s" % e)
+        return None
+
+
+def validate_keys_dataset(dataset):
+    """
+    This method validates the JSON object to check if it contains all the mandatory keys.
+    """
+    isValid = True
+    missing_keys = []
+
+    if "survey_id" not in dataset.keys():
+        missing_keys.append("survey_id")
+    if "period_id" not in dataset.keys():
+        missing_keys.append("period_id")
+    if "sds_schema_version" not in dataset.keys():
+        missing_keys.append("sds_schema_version")
+    if "schema_version" not in dataset.keys():
+        missing_keys.append("schema_version")
+    if "form_type" not in dataset.keys():
+        missing_keys.append("form_type")
+
+    if len(missing_keys) > 0:
+        isValid = False
+
+    return isValid, missing_keys
```

Comment:
Please could you write tests for the validation logic here? Ideally for each key we're validating plus the new logic added in get_dataset

Reply:
@JackRimmerAND Same as above - added a test in 'dataset_storage_test.py' specifically for the 'validate_keys' method.